utils/docker/Dockerfile.leap.15

```diff
@@ -136,9 +136,24 @@ RUN [ "$DAOS_DEPS_BUILD" != "yes" ] || \
     { scons --build-deps=yes --jobs $DEPS_JOBS PREFIX=/opt/daos \
       TARGET_TYPE=$DAOS_TARGET_TYPE --deps-only && \
     ([ "$DAOS_KEEP_BUILD" != "no" ] || /bin/rm -rf build *.gz); }
+USER root:root
+
+# Load the COMPILER arg early, and optionally install the Intel compiler
+# Do this before the CACHEBUST to make best use of cache as this is a big
+# download, and make it optional on the value of COMPILER for the same
```

Comment:
It's a shame it is so big.  I wonder how much of that big download is actually only necessary to do compilation and how much extra baggage is being carried in the way of developer conveniences, etc.

But looking at the run, it only took a small number of minutes.  That doesn't seem too bad.  Is it much worse in GHA?

This still we be low hanging fruit as more stages that **must** run in Jenkins need to be added.

Reply:
There's nothing to say we couldn't do it in GHA, but it would add a considerable amount of extra data and cycles per PR, and it feels like we're already falling behind in GHA recently.  If there was a better way of using the docker cache it would really help, hopefully with self-hosted runners this whole problem will disappear.

Reply:
Yeah.  Self-hosted runners FTW.

I'm frankly surprised that GH are not doing more/better at caching Docker images.  They have to realize how much resource is being wasted building images from scratch every time.  Surely that waste is worth investing in being able to cache at least the most frequently used images/layers.

Given that we already have all of the Docker infrastructure up and running, I wonder how much/little effort gatewaying GHA to it through a self-hosted runner actually is.  I have set up a runner before.  It's pretty trivial.  The only (small, hopefully) hurdle beyond that is to make the runner shovel off jobs to the Docker host.  Maybe we just put the runner right on our Docker hosts (wolf-3[3-4]).